lib/iris/coords.py

```diff
@@ -1628,6 +1628,214 @@ def __hash__(self):
+
+        """
+        # Turn the key(s) into a full slice spec - i.e. one entry for
+        # each dimension of the cell_measure.
+        full_slice = iris.util._build_full_slice_given_keys(key, self.ndim)
+
+        # If it's a "null" indexing operation (e.g. cell_measure[:, :]) then
+        # we can preserve deferred loading by avoiding promoting _data
+        # and _bounds to full ndarray instances.
+        def is_full_slice(s):
+            return isinstance(s, slice) and s == slice(None, None)
+        if all(is_full_slice(s) for s in full_slice):
+            data = self._data
+        else:
+            data = self._data
+
+            # Make indexing on the cube column based by using the
+            # column_slices_generator (potentially requires slicing the
+            # data multiple times).
+            _, slice_gen = iris.util.column_slices_generator(full_slice,
+                                                             self.ndim)
+            for keys in slice_gen:
+                if data is not None:
+                    data = data[keys]
+                    if data.shape and min(data.shape) == 0:
+                        raise IndexError('Cannot index with zero length '
+                                         'slice.')
+
+        new_cell_measure = self.copy(data=data)
+        return new_cell_measure
+
+    def copy(self, data=None):
+        """
+        Returns a copy of this CellMeasure.
+
+        Kwargs:
+
+        * data: A data array for the new cell_measure.
+                This may be a different shape to the data of the
+                cell_measure being copied.
+
+        """
+        new_cell_measure = copy.deepcopy(self)
+        if data is not None:
+            # Explicitly not using the data property as we don't want the
+            # shape the new data to be constrained by the shape of
+            # self.data
+            new_cell_measure._data = None
+            new_cell_measure.data = data
+
+        return new_cell_measure
+
+    def _repr_other_metadata(self):
+        fmt = ''
+        if self.long_name:
+            fmt = ', long_name={self.long_name!r}'
+        if self.var_name:
+            fmt += ', var_name={self.var_name!r}'
+        if len(self.attributes) > 0:
+            fmt += ', attributes={self.attributes}'
+        result = fmt.format(self=self)
+        return result
+
+    def __str__(self):
+        result = repr(self)
+        return result
+
+    def __repr__(self):
+        fmt = ('{cls}({self.data!r}'
+               ', measure={self.measure}, standard_name={self.standard_name!r}'
+               ', units={self.units!r}{other_metadata})')
+        result = fmt.format(self=self, cls=type(self).__name__,
+                            other_metadata=self._repr_other_metadata())
+        return result
+
+    def _as_defn(self):
+        defn = CoordDefn(self.standard_name, self.long_name, self.var_name,
+                         self.units, self.attributes, self.measure)
```

Comment:
@marqh This is a bit naughty! 

You're hijacking `CoordDefn` and all the actual parameters to formal parameters align apart from the actual parameter `self.measure` which is being mapped to the formal parameter `coord_system` in `CoordDefn` ... :bomb: